### PR TITLE
Enable ccache for SRN PR testing

### PR DESCRIPTION
**Pull request by jmlapre**
@trilinos/framework 

## Motivation

As anyone associated with Trilinos, I want the autotester to be faster.

Use ccache for PR testing.  Ideally, start with the CUDA build, since it takes the longest.

Store the cache at `/fgs/trilinos/ccache/cache`.

This is a follow-up to #11386.

https://sems-atlassian-son.sandia.gov/jira/browse/TRILFRAME-521

### CODE REVIEW

**Comment by sebrowne** [comment]
Should this also set CCACHE_NODISABLE to engage it in the CMake code from #11386?

**Comment by jmlapre** [comment]
> Should this also set CCACHE_NODISABLE to engage it in the CMake code from #11386?

Yes, good catch.

File: packages/framework/pr_tools/PullRequestLinuxDriver.sh [hunk 1]
```diff
@@ -8,6 +8,16 @@ source ${SCRIPTPATH:?}/common.bash
 on_weaver=$(echo "$@" | grep '\-\-on_weaver' &> /dev/null && echo "1")
 on_ats2=$(echo "$@" | grep '\-\-on_ats2' &> /dev/null && echo "1")
 
+# Configure ccache via environment variables
+function configure_ccache() {
+    print_banner "Configuring ccache"
+
+    envvar_set_or_create CCACHE_NODISABLE true
+    envvar_set_or_create CCACHE_BASEDIR '/fgs/trilinos/ccache/cache'
```
Comment (sebrowne) [inline packages/framework/pr_tools/PullRequestLinuxDriver.sh:16]: Sorry I missed this before, but I think what we want is this:
CCACHE_DIR '/fgs/trilinos/ccache/cache'
CCACHE_BASEDIR '/path/to/trilinos/source/code'

I imagine that second one is available as a variable in here somewhere.

**Review by sebrowne — CHANGES_REQUESTED** [review]

**Review by sebrowne — APPROVED** [review]

**Comment by jmlapre** [comment]
Unfortunately it seems to be a greater possibility than anticipated:
  
https://gitlab.kitware.com/cmake/cmake/-/issues/24345

I think ccache is likely to be a bigger win than ninja but we'll wait and see the results...

**Comment by sebrowne** [comment]
> Unfortunately it seems to be a greater possibility than anticipated:
> 
> https://gitlab.kitware.com/cmake/cmake/-/issues/24345
> 
> I think ccache is likely to be a bigger win than ninja but we'll wait and see the results...

@jmlapre I think this only applies to coverage builds (.gcno files are only generated for coverage-profiling enabled builds as far as I know)

**Comment by sebrowne** [comment]
Okay, I did some diagnosis on this today.  I think that what is happening is that Trilinos (nearly every source file) includes generated headers, which means it has -I directives that point into the BUILD directory.  For pretty obvious reasons, this makes it impossible to cache files on a per-object basis.  Will discuss at standup tomorrow.

**Comment by sebrowne** [comment]
Closing since #11616 is now merged

**Pull request closed without merging**